plugins/inputs/t128_graphql/t128_graphql.go

```diff
@@ -0,0 +1,284 @@
+package t128_graphql
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Jeffail/gabs"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	//make this configurable?
+	DefaultRequestTimeout = time.Second * 5
+)
+
+type T128GraphQL struct {
+	CollectorName string            `toml:"collector_name"`
+	BaseURL       string            `toml:"base_url"`
+	EntryPoint    string            `toml:"entry_point"`
+	Fields        map[string]string `toml:"extract_fields"`
+	Tags          map[string]string `toml:"extract_tags"`
+
+	Client *http.Client
+	Query  string
+}
+
+var sampleConfig = `
+## Collect data using graphQL
+[[inputs.t128_graphql]]
+## Required. The telegraf collector name
+# collector_name = "arp-state"
+
+## Required. GraphQL ports vary across 128T versions
+# base_url = "http://localhost:31517"
+
+## The starting point in the graphQL tree for all configured tags and fields
+# entry_point = "allRouters[name:RTR_WEST_COMBO]/nodes/nodes[name:combo-west]/nodes/arp/nodes"
+
+## The fields to collect with the desired name as the key (left) and graphQL 
+## key as the value (right)
+# [inputs.t128_graphql.extract_fields]
```

Comment:
This config seems to describe a single metric. I'm wondering whether it would be worth it to be able to describe multiple metrics in one config? I think you would still be able to put multiple of these `[[inputs.t128_graphql]]` in a single input file. So I guess the only savings you would get would be the savings of only having to specify the base_url once and initializing everything here once. Doesn't really seem worth it, but just a thought I had.

Reply:
Right that would make it closer to t128_metrics. I like the idea and will add that in `advanced_syntax` that will be covered if time permits